waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/model/QueryMapping.java

```diff
@@ -15,13 +15,16 @@
  */
 package com.hotels.bdp.waggledance.mapping.model;
 
+import com.hotels.bdp.waggledance.api.WaggleDanceException;
+
 public interface QueryMapping {
 
   /**
    * @param metaStoreMapping
    * @param query
    * @return query where database names are correctly mapped using metaStoreMapping. example 'select id from db.table'
    *         -> 'select id from waggle_db.table'
+   * @throws WaggleDanceException when the transform could not be done.
    */
-  String transformOutboundDatabaseName(MetaStoreMapping metaStoreMapping, String query);
+  String transformOutboundDatabaseName(MetaStoreMapping metaStoreMapping, String query) throws WaggleDanceException;
```

Comment:
is `WaggleDanceException` not a `RuntimeException`?

Reply:
it is but I wanted to make it part of the Interface contract so users of it know what can be expected to catch

Reply:
is documented, it doesn't need to be in the method signature